relay/relay.go

```diff
@@ -262,15 +270,16 @@ func (r *Relay) addPeer(peerID core.PeerID) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	// TODO: account for peers that rejoin
 	if len(r.peers) == 0 || r.peers[len(r.peers)-1] < peerID {
 		r.peers = append(r.peers, peerID)
 		return
 	}
 
 	// TODO: can optimize via binary search
 	for index, peer := range r.peers {
-		if peer > peerID {
+		if peer == peerID {
```

Comment:
This function has to be refactored. Why break ? Do a simple search and add if it does not exist.

Reply:
Not sure I follow, we should also avoid inserting duplicate peers (like a relay joins -> disconnects and the joins again).